### ISSUE TITLE
bug(watcher): changes on a file newly added to dependency are not watched

### ISSUE THREAD

**Issue by magurotuna** [denoland/deno#8181]
### version

current latest master (bfce376c)

### How to reproduce

1. create a file named `main.js`with the following content:

```js
console.log('hello');
```
2. start running deno watcher by `cargo run -- run --watch --unstable main.js`. At this point we can see an output like:

```
hello
Watcher Process terminated! Restarting on file change...
```
3. create a new file named `new.js`. Nothing happens on the terminal.

```js
export const world = 'world';
```

4. edit `main.js` to the following and save it:

```diff
+import { world } from "./new.js";
console.log("hello");
+console.log(world);
```

When saving it, we get:

```
Watcher File change detected! Restarting!
hello
world
Watcher Process terminated! Restarting on file change...
```

5. edit the value of `world` in `new.js` and save it:

```diff
-export const world = "world";
+export const world = "Deno!";
```

When saving it, nothing happens although we expect `hello\nDeno!` to be printed.

I've already found the cause of this problem in the watcher implementation. This is because module resolution is done only once at the starting of the watcher. Basically we should resolve module every time any event is detected.

I'll look into this problem deeply and try to elabolate a solution.

**Comment by magurotuna** [comment]
oops, this seems duplicate of #8035 
I'd like to continue it here as I wrote more detailed situation.
